File: automation/lib/berlin-flats-summary.ts
#!/usr/bin/env bun

type HuntListing = {
  title?: string | null;
  cold_rent?: number | null;
  district?: string | null;
  verdict?: string | null;
  url?: string | null;
};

type HuntOutput = {
  new?: HuntListing[];
};

const input = await Bun.stdin.text();
let data: HuntOutput;
try {
  data = JSON.parse(input) as HuntOutput;
} catch {
  process.exit(0);
}

const listings = (data.new ?? []).filter((listing) => listing.verdict === "pending" || listing.verdict === "review");
if (listings.length === 0) process.exit(0);

const lines = [`🏠 ${listings.length} new Berlin flat(s):`];
for (const listing of listings.slice(0, 10)) {
  const rent = listing.cold_rent ?? "?";
  const verdict = listing.verdict === "review" ? "⚠️ review" : "✓";
  lines.push(`• ${listing.title || "(no title)"} - €${rent}, ${listing.district || "?"} [${verdict}]\n${listing.url || ""}`);
}

process.stdout.write(`${lines.join("\n")}\n`);

File: automation/jobs/berlin-flats-summary.py
#!/usr/bin/env python3
"""Summarize `hunt.js --json` output for Telegram notification.

Reads the hunt JSON on stdin and prints a short summary of new listings whose
verdict is `pending` or `review`. Prints nothing (exit 0) when there is nothing
actionable or the input can't be parsed, so the caller sends no notification.
"""
import sys
import json

try:
    data = json.load(sys.stdin)
except Exception:
    sys.exit(0)

new = [l for l in data.get("new", []) if l.get("verdict") in ("pending", "review")]
if not new:
    sys.exit(0)

lines = [f"\U0001F3E0 {len(new)} new Berlin flat(s):"]
for l in new[:10]:
    rent = l.get("cold_rent") or "?"
    verdict = "⚠️ review" if l.get("verdict") == "review" else "✓"
    title = l.get("title") or "(no title)"
    district = l.get("district") or "?"
    url = l.get("url") or ""
    lines.append(f"• {title} — €{rent}, {district} [{verdict}]\n{url}")
print("\n".join(lines))
